src/action/list_directory.rs

```diff
@@ -0,0 +1,727 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! A handler and associated types for the list directory action.
+//!
+//! A list directory action stats all files in the provided directory.
+
+use crate::session::{self, Session};
+use rrg_proto::{ListDirRequest, StatEntry, path_spec::PathType};
+
+use std::fs::{self, Metadata};
+use std::path::PathBuf;
+use std::fmt::{Display, Formatter};
+use log::warn;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+#[cfg(target_os = "linux")]
+use std::time::Duration;
+
+#[cfg(target_os = "linux")]
+use std::fs::File;
+
+#[cfg(target_os = "linux")]
+use ioctls;
+
+#[cfg(target_os = "linux")]
+use std::os::raw::c_long;
+
+#[cfg(target_os = "linux")]
+use std::os::unix::fs::MetadataExt;
+
+#[cfg(target_os = "linux")]
+use std::os::unix::io::AsRawFd;
+
+#[derive(Debug)]
+enum Error {
+    ReadPath(std::io::Error),
+}
+
+impl std::error::Error for Error {
+
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        use Error::*;
+
+        match *self {
+            ReadPath(ref error) => Some(error),
+        }
+    }
+}
+
+impl Display for Error {
+
+    fn fmt(&self, fmt: &mut Formatter) -> std::fmt::Result {
+        use Error::*;
+
+        match *self {
+            ReadPath(ref error) => {
+                write!(fmt, "unable to read path: {}", error)
+            }
+        }
+    }
+}
+
+impl From<Error> for session::Error {
+
+    fn from(error: Error) -> session::Error {
+        session::Error::action(error)
+    }
+}
+
+#[derive(Debug)]
+enum ParseError {
+    UnsupportedValue(String, String),
+}
+
+impl std::error::Error for ParseError {
+
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        use ParseError::*;
+
+        match *self {
+            UnsupportedValue(ref _field, ref _value) => None,
```

Comment:
You can just use a wildcard (`_`, no need for names).

Reply:
Fixed.